Source/DafnyLanguageServer.Test/Lookup/DefinitionTest.cs

```diff
@@ -28,20 +25,265 @@ public class DefinitionTest : ClientBasedLanguageServerTest {
     }
 
     [TestMethod]
-    public async Task DefinitionOfMethodInvocationOfMethodDeclaredInSameDocumentReturnsLocation() {
+    public async Task WhileLoop() {
       var source = @"
-method DoIt() returns (x: int) {
+method HasLoop() {
+  var x := 1;
+  while(true) {
+    if (x > 2) {
+      break;
+    }
+    x := x + 1;
+  }
 }
+".TrimStart();
+
+      var documentItem = CreateTestDocument(source);
+      await client.OpenDocumentAndWaitAsync(documentItem, CancellationToken);
+      var whileReference = (await RequestDefinition(documentItem, (4, 8)).AsTask()).Single();
+      Assert.AreEqual(new Range((2, 2), (2, 7)), whileReference.Location!.Range);
```

Comment:
Thank you for adding these tests! I will surely make go to definition more robust in the long term.

I have another idea. Although it is more work for you, I think it will make it easier to read and maintain those tests.  Let me know if you like it.

Let's say you define the following

```cs
var sourceAndRequests = @"
method HasLoop() {
  var x := 1;
//^^^^^=1
  while(true) {
    if (x > 2) {
//      ^?1
      break;
    }
    x := x + 1;
  }
}
".TrimStart();
```

Then you could extract the source code, the request, and the expected answer directly from the same source. Of course you have to fix the line numbers when extracting everything, but to alleviate this concern, you can borrow some of the code for the hover tests which define tests this way:

https://github.com/dafny-lang/dafny/blob/62a6d582fbed84e405952483accbf095933c1acb/Source/DafnyLanguageServer.Test/Lookup/HoverTest.cs#L78-L85

and extract it like this:

https://github.com/dafny-lang/dafny/blob/62a6d582fbed84e405952483accbf095933c1acb/Source/DafnyLanguageServer.Test/Lookup/HoverTest.cs#L58-L72

I know it's a bit challenging and you might have other tasks to do though.

Reply:
> However, I'd prefer making such a change in a separate PR. I can do it as a follow-up to this one and use it in the DefinitionTest.cs file.

Yes I would really appreciate a follow-up PR for that.

Reply:
It's good that you call it out :-) I've put it on my list, thanks.